### PR TITLE
Use clock_gettime instead of gettimeofday, it's more efficiency on low power system

### PR DESCRIPTION
**Pull request by Mullin**
Use clock_gettime instead of gettimeofday, it's more efficiency on low power system

### CODE REVIEW

**Comment by tueidj** [comment]
Disregarding portability issues (I don't believe OSX has clock_gettime()), this is actually a good idea on Android; it prevents the clock jumping around if the system is set to update automatically from the network and the device switches between cell towers that aren't perfectly synchronized.

**Comment by Mullin** [comment]
Yeah, i know. I just forgot to add a def for OSX. I'm a bit busy. I commit soon as possible :)

**Comment by skidau** [comment]
Please fix up the commit history.

**Comment by delroth** [comment]
Definitely agree about using clock_gettime and CLOCK_MONOTONIC being superior, but I really doubt this is any different in terms of efficiency. In fact, I would expect gettimeofday to be faster on x86_64 according to the VDSO implementation (gettimeofday is implemented as do_realtime + a division, while clock_gettime needs to also branch depending on the clock type).

File: Source/Core/Common/Timer.cpp [hunk 1]
```diff
@@ -48,10 +52,14 @@ u64 Timer::GetTimeUs()
 	static double freq = GetFreq();
 	QueryPerformanceCounter(&time);
 	return u64(double(time.QuadPart) * freq);
-#else
+#elif defined __APPLE__
 	struct timeval t;
 	(void)gettimeofday(&t, nullptr);
-	return ((u64)(t.tv_sec * 1000000 + t.tv_usec));
+	return ((u64)(t.tv_sec * 1000 + t.tv_usec / 1000));
```
Comment (degasus) [inline Source/Core/Common/Timer.cpp]: This line is likely wrong

**Comment by degasus** [comment]
@dolphin-emu-bot rebuild

File: Source/Core/Common/Timer.cpp [hunk 2]
```diff
@@ -219,9 +231,12 @@ double Timer::GetDoubleTime()
 #ifdef _WIN32
 	struct timeb tp;
 	(void)::ftime(&tp);
-#else
-	struct timeval t;
+#elif defined __APPLE__
+	struct timespec t;
```
Comment (degasus) [inline Source/Core/Common/Timer.cpp]: more likely timeval

**Comment by Mullin** [comment]
Hmmm.. So just defined clockgettime for ARM devices and x86 and leave gettimeofday for x86-64

**Comment by delroth** [comment]
Don't, nobody cares about its efficiency anyway (I can't imagine a situation where it would be the bottleneck). CLOCK_MONOTONIC has clear advantages.

**Comment by comex** [comment]
FYI, you can use `mach_absolute_time` for this purpose on OS X (it is monotonic and fast).  See:

http://shiftedbits.org/2008/10/01/mach_absolute_time-on-the-iphone/

**Comment by Mullin** [comment]
@skidau history fixed.

**Comment by magumagu** [comment]
@dolphin-emu-bot rebuild

File: Source/Core/Common/Timer.cpp [hunk 3]
```diff
@@ -237,7 +252,7 @@ double Timer::GetDoubleTime()
 #ifdef _WIN32
 	double ms = tp.millitm / 1000.0 / 1000.0;
 #else
```
Comment (degasus) [inline Source/Core/Common/Timer.cpp]: #elif defined __APPLE__

**Comment by Mullin** [comment]
Benchmark : 
BUST A MOVE 3000 before fix : 16-17 fps
BUST A MOVE 3000 after fix : 30-34 fps

**Comment by Mullin** [comment]
@dolphin-emu-bot rebuild

**Comment by degasus** [comment]
@dolphin-emu-bot rebuild

**Comment by degasus** [comment]
I'd like to see those commits squashed, else LGTM

**Comment by Mullin** [comment]
Squashed :)

**Comment by degasus** [comment]
@dolphin-emu-bot rebuild
LGTM

**Pull request merged**